spec/index.bs

```diff
@@ -3646,6 +3667,24 @@ dictionary GPUOrigin3DDict {
 typedef (sequence<GPUIntegerCoordinate> or GPUOrigin3DDict) GPUOrigin3D;
 </script>
 
+An <dfn dfn>Origin3D</dfn> is a {{GPUOrigin3D}}.
+[=Origin3D=] is a spec namespace for the following definitions:
+<!-- This is silly, but provides convenient syntax for the spec. -->
```

Comment:
what's silly is that we allow both sequence and dictionary representation here, even though sequences are more heavy and can be trivially converted to a dictionary on the user side. This whole section wouldn't need to exist.

Reply:
Why are sequences more heavy? On the syntax side they're much lighter at least.

As you know I still want to accept only sequence here, but if there's a strong preference we can change it back to dictionary-only. FWIW at some point I do want to consider making it so different users of this type have different defaults (e.g. some require all 3, some require only 2 or 1).

Reply:
AFAIK, because sequence is a separate object in JS. For this reason we wanted to avoid them in `setVertexBuffers`, for example.

Reply:
A dictionary is also a separate object though. We would avoid a dictionary in setVertexBuffers too.

Reply:
Indeed! Would there be a difference when coming from WASM though?

Reply:
Yes actually, I keep forgetting about this. `sequence` allows a typed array, and actually I think any iterable (I just tried it with a Float32Array, and a generator `(function*() { ... })()` and those worked. Set didn't work but maybe it's supposed to.)

Reply:
So would it be correct to say that `sequence` is heavier for WASM case?

Reply:
I'd say it's slightly lighter. For WASM you can do any of:

- `{ x: HEAPF32[i], y: HEAPF32[i + 1], z: HEAPF32[i + 2] }` copy and construct dict
- `[ HEAPF32[i], HEAPF32[i + 1], HEAPF32[i  + 2] ]` copy and construct array
- `HEAPF32.slice(i, i + 3)` create new Float32Array + ArrayBuffer and memcpy
- `HEAPF32.subarray(i, i + 3)` create new Float32Array pointing to same ArrayBuffer